### PR TITLE
Spectating with bucket

### PR DESCRIPTION
**Pull request by Dev-Shaz**

### CODE REVIEW

File: scripts/menu/server/sv_player_modal.lua [hunk 1]
```diff
@@ -68,9 +68,25 @@ RegisterNetEvent('txAdmin:menu:spectatePlayer', function(id)
     if not target then
       return
     end
+    local tgtBucket = GetPlayerRoutingBucket(id)
+    local srcBucket = GetPlayerRoutingBucket(src)
+    if tgtBucket ~= srcBucket then 
+      ADMIN_DATA[tostring(src)].bucket = srcBucket
```
Comment (TasoOneAsia) [inline scripts/menu/server/sv_player_modal.lua:74]: I feel like this isn't the best place to store the data regarding bucket as we haven't traditionally used this map for non-admin related data but meh.

An alternative is we can store the original bucket on the calling client.

File: scripts/menu/server/sv_player_modal.lua [hunk 2]
```diff
@@ -68,9 +68,25 @@ RegisterNetEvent('txAdmin:menu:spectatePlayer', function(id)
...
 
     local tgtCoords = GetEntityCoords(target)
     TriggerClientEvent('txAdmin:menu:specPlayerResp', src, id, tgtCoords)
   end
   TriggerEvent('txaLogger:menuEvent', src, 'spectatePlayer', allow, id)
+end)
+
+
+
+RegisterNetEvent('txAdmin:menu:endSpectate', function()
```
Comment (TasoOneAsia) [inline scripts/menu/server/sv_player_modal.lua:86]: This net event can be used to spam errors or change an admins bucket without any verification. Please run verification checks on the calling source using the HasPermission function and `players.spectate`

**Review by TasoOneAsia — CHANGES_REQUESTED** [review]

File: scripts/menu/server/sv_player_modal.lua (hunk 2 above)
Comment (Dev-Shaz) [reply scripts/menu/server/sv_player_modal.lua:86]: Will do my bad

Comment (Dev-Shaz) [reply scripts/menu/server/sv_player_modal.lua:86]: Done

**Review by TasoOneAsia — APPROVED** [review]

**Pull request merged**